browser/brave_stats_updater.cc

```diff
@@ -191,8 +244,12 @@ void BraveStatsUpdater::SendServerPing() {
   auto resource_request = std::make_unique<network::ResourceRequest>();
   auto stats_updater_params =
       std::make_unique<brave::BraveStatsUpdaterParams>(pref_service_);
-  resource_request->url =
-      GetUpdateURL(GURL(g_base_update_url_), *stats_updater_params);
+
+  auto endpoint =
+      GURL(BuildStatsEndpoint((threshold_ping_state_ == THRESHOLD_PING_FIRED) ?
```

Comment:
if you're building this dynamically I would add a DCHECK to make sure it's a valid gurl